src/vs/workbench/api/browser/mainThreadEditor.ts

```diff
@@ -148,6 +153,21 @@ export class MainThreadTextEditorProperties {
 			&& a.insertSpaces === b.insertSpaces
 			&& a.cursorStyle === b.cursorStyle
 			&& a.lineNumbers === b.lineNumbers
+			&& equals(a.rulers, b.rulers, (a, b) => {
```

Comment:
This `equals` function is duplicated in another file below, but I wasn't really sure where to put it if I used a common file.